tests/e2e/vsan_stretched_cluster_utils.go

```diff
@@ -246,3 +248,111 @@ func createPvcInParallel(client clientset.Interface, namespace string, diskSize
+			return true, nil
+		}
+		return false, nil
+	})
+	return waitErr
+}
+
+// runCmdonESXInParallel runs command on multiple ESX parallelly
+func runCmdOnHostsInParallel(hostIP string, sshCmd string, wg *sync.WaitGroup) {
+	defer wg.Done()
+	op, err := runCommandOnESX("root", hostIP, sshCmd)
+	framework.Logf(op)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+}
+
+// networkFailureParallel causes or removes network failure on a particular site
+func networkFailureParallel(hosts []string, causeNetworkFailure bool) {
+	var wg sync.WaitGroup
+	if causeNetworkFailure {
+		framework.Logf("Creating a Network Failure")
+		sshCmd := "localcli network firewall set --enabled true;"
+		sshCmd += "localcli network firewall ruleset set --allowed-all 0 --ruleset-id cmmds;"
+		sshCmd += "localcli network firewall ruleset set --allowed-all 0 --ruleset-id rdt;"
+		sshCmd += "localcli network firewall ruleset set --allowed-all 0 --ruleset-id fdm;"
+
+		wg.Add(len(hosts))
+		for _, host := range hosts {
+			go runCmdOnHostsInParallel(host, sshCmd, &wg)
+		}
+		wg.Wait()
+		sshCmd = "vsish -e set /vmkModules/esxfw/globaloptions 1 0 0 0 1"
+		wg.Add(len(hosts))
+		for _, host := range hosts {
+			go runCmdOnHostsInParallel(host, sshCmd, &wg)
+		}
+		wg.Wait()
+	} else {
+		framework.Logf("Removing network Failure")
+		sshCmd := "localcli network firewall set --enabled false;"
+		sshCmd += "localcli network firewall ruleset set --allowed-all 1 --ruleset-id cmmds;"
+		sshCmd += "localcli network firewall ruleset set --allowed-all 1 --ruleset-id rdt;"
+		sshCmd += "localcli network firewall ruleset set --allowed-all 1 --ruleset-id fdm;"
+
+		wg.Add(len(hosts))
+		for _, host := range hosts {
+			go runCmdOnHostsInParallel(host, sshCmd, &wg)
+		}
+		wg.Wait()
+		sshCmd = "vsish -e set /vmkModules/esxfw/globaloptions 1 1 0 1 1"
+		wg.Add(len(hosts))
+		for _, host := range hosts {
+			go runCmdOnHostsInParallel(host, sshCmd, &wg)
+		}
+		wg.Wait()
+	}
+}
+
+// deletePVCInParallel deletes PVC in a given namespace in parallel
+func deletePvcInParallel(client clientset.Interface, pvclaims []*v1.PersistentVolumeClaim,
+	namespace string, wg *sync.WaitGroup) {
+	defer wg.Done()
+	for _, pvclaim := range pvclaims {
+		err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	}
+}
+
+// createPodsInParallel creates Pods in a given namespace in parallel
+func createPodsInParallel(client clientset.Interface, namespace string, pvclaims []*v1.PersistentVolumeClaim,
+	lock *sync.Mutex, ch chan *v1.Pod, wg *sync.WaitGroup, volumeOpsScale int) {
+	defer wg.Done()
+	for i := 0; i < volumeOpsScale; i++ {
+		pod := fpod.MakePod(namespace, nil, []*v1.PersistentVolumeClaim{pvclaims[i]}, false, execCommand)
+		pod.Spec.Containers[0].Image = busyBoxImageOnGcr
+		pod, err := client.CoreV1().Pods(namespace).Create(context.TODO(), pod, metav1.CreateOptions{})
```

Comment:
you can pass `ctx` as a param to this method

Reply:
Added ctx param to createPodsInParallel method